internal/controllers/clusterextension_controller.go

```diff
@@ -155,6 +167,7 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 		ext.Status.InstalledBundle = nil
 		setInstalledStatusConditionFailed(&ext.Status.Conditions, err.Error(), ext.GetGeneration())
 		setDeprecationStatusesUnknown(&ext.Status.Conditions, "deprecation checks have not been attempted as installation has failed", ext.GetGeneration())
+		setProgressingStatusConditionProgressing(&ext.Status.Conditions, "installation failed", ext.GetGeneration())
```

Comment:
If installed is explicitly set to failed, and we are not able to find if the bundle deployment exists on cluster or not, I think this should be set to `False` instead, since there is no other state of improvement after this. Was there a specific reason to make it true instead?

Reply:
I thought of the progressing condition as more of an indication of  'an install/upgrade is in progress, the currently installed bundle may be unstable'. Since it's stuck progressing here, it should be an indicator that the current installation is possibly incomplete/broken.